src/components/structures/RoomSubList.js

```diff
@@ -100,9 +100,11 @@ const RoomSubList = React.createClass({
 
     applySearchFilter: function(list, filter) {
         if (filter === "") return list;
-        return list.filter((room) => {
-            return room.name && room.name.toLowerCase().indexOf(filter.toLowerCase()) >= 0;
-        });
+        const lcFilter = filter.toLowerCase();
+        // case insensitive if room name includes filter,
+        // or if starts with `#` and one of room's aliases starts with filter
+        return list.filter((room) => (room.name && room.name.toLowerCase().includes(lcFilter)) ||
+            (filter[0] === '#' && room.getAliases().some((alias) => alias.toLowerCase().startsWith(lcFilter))));
```

Comment:
This doesn't look like a merge commit, it includes changes to include matches against room aliases. 

That's fine, but let's squash this PR (when merging) for simplicity.

Reply:
https://github.com/matrix-org/matrix-react-sdk/blob/develop/src/components/structures/RoomSubList.js#L111

Reply:
https://github.com/matrix-org/matrix-react-sdk/pull/1957/files